### DISCUSSION TITLE
generic message decoding

### DISCUSSION THREAD

**Discussion by mattwyd** [xR3b0rn/dbcppp#152] [General]
how would you recommend I decode a given can frame when i don't necessarily know the name I only get the CAN ID and 8 BYTES of data on the bus. doing a big case statement seems ugly to me

**Comment by xR3b0rn** [comment]
1. Create your hash-map mapping from CAN ID to message pointer (dbcppp inernally uses a map as well, however the implementation of dbcppp is rather slow)
2. Whenever, you receive a frame, lookup the map and use the decode function
